### ISSUE TITLE
ant with failOnViolation="false" will FAIL

### ISSUE THREAD

**Issue by hchan** [checkstyle/checkstyle#2185]
http://checkstyle.sourceforge.net/anttask.html

If I use 
```xml
<target name="checkstyle"
	        description="Generates a report of code convention violations."
		>
		<!-- See http://checkstyle.sourceforge.net/anttask.html for full options of using checkstyle with ant-->
		<checkstyle config="checkstyle.xml"
	              failOnViolation="false"
			>
			<formatter type="xml" tofile="checkstyle_report.xml"/>
			<fileset dir="src" >
				<include name="**/*.java"/>
			</fileset>
		</checkstyle>
	</target>

and run 
ant checkstyle
I get:
BUILD FAILED

**Comment by romani** [comment]
I do not see violation messages in output, and this option is failing build if violation present.

Please share more information/context for this issue

**Comment by hchan** [comment]
FailsOnViolation (expected) on version 6.9 but not 6.10

**Comment by romani** [comment]
example projects to start testing https://github.com/sevntu-checkstyle/checkstyle-samples/tree/master/ant-project

**Comment by romani** [comment]
in addtion:

1)
http://checkstyle.sourceforge.net/anttask.html#Description

 please remove "The task has been tested using ANT 1.5." it work fine on recent version too
```
/var/temp/sevntu/checkstyle-samples/ant-project [master|✔] $ ant -v
Apache Ant(TM) version 1.8.2 compiled on December 3 2011
```

2)
http://checkstyle.sourceforge.net/anttask.html#Examples

above all , please add link to https://github.com/sevntu-checkstyle/checkstyle-samples/tree/master/ant-project , real example is always better then some config chunks.

**Comment by Vladlis** [comment]
@hchan 
Can you provide a full log? I suppose that the `taskdef` was not set properly.

**Comment by romani** [comment]
@Vladlis , if you can not reproduce this issue, please prove that it works correctly by cli on test project (in sevntu repo) . If all works as required we will close the issue

**Comment by Vladlis** [comment]
@romani 
In test project everything works correctly. Tested under Ant 1.9

**Comment by romani** [comment]
@Vladlis ,

>  please prove that it works correctly by cli on test project (in sevntu repo)

prove it by console output.

**Comment by Vladlis** [comment]
```
vlad@vlad-N61PC-M2S ~/projects/checkstyle-samples/ant-project $ ant -v
Apache Ant(TM) version 1.9.3 compiled on April 8 2014
```
With failOnViolation="false"
```
vlad@vlad-N61PC-M2S ~/projects/checkstyle-samples/ant-project $ ant checkstyle
Buildfile: /home/vlad/projects/checkstyle-samples/ant-project/build.xml

download-ivy:
      [get] Getting: http://repo2.maven.org/maven2/org/apache/ivy/ivy/2.1.0-rc2/ivy-2.1.0-rc2.jar
      [get] To: /home/vlad/projects/checkstyle-samples/ant-project/lib/ivy.jar
      [get] Not modified - so not downloaded

init-ivy:

resolve:
[ivy:retrieve] :: Ivy 2.1.0-rc2 - 20090704004254 :: http://ant.apache.org/ivy/ ::
[ivy:retrieve] :: loading settings :: file = /home/vlad/projects/checkstyle-samples/ant-project/ivysettings.xml
[ivy:retrieve] :: resolving dependencies :: com.github.sevntu.checkstyle#sample;working@vlad-N61PC-M2S
[ivy:retrieve] 	confs: [default]
[ivy:retrieve] 	found com.puppycrawl.tools#checkstyle;6.15 in local-maven-2
[ivy:retrieve] 	found org.apache.commons#commons-lang3;3.4 in local-maven-2
[ivy:retrieve] 	found antlr#antlr;2.7.7 in local-maven-2
[ivy:retrieve] 	found org.antlr#antlr4-runtime;4.5.1-1 in local-maven-2
[ivy:retrieve] 	found commons-beanutils#commons-beanutils;1.9.2 in local-maven-2
[ivy:retrieve] 	found commons-logging#commons-logging;1.1.1 in local-maven-2
[ivy:retrieve] 	found commons-collections#commons-collections;3.2.1 in local-maven-2
[ivy:retrieve] 	found commons-cli#commons-cli;1.3.1 in local-maven-2
[ivy:retrieve] 	found com.google.guava#guava;19.0 in local-maven-2
[ivy:retrieve] 	found org.apache.ant#ant;1.9.6 in local-maven-2
[ivy:retrieve] 	found org.apache.ant#ant-launcher;1.9.6 in local-maven-2
[ivy:retrieve] 	found com.github.sevntu.checkstyle#sevntu-checks;1.18.0 in local-maven-2
[ivy:retrieve] :: resolution report :: resolve 367ms :: artifacts dl 17ms
[ivy:retrieve] 	:: evicted modules:
[ivy:retrieve] 	com.puppycrawl.tools#checkstyle;6.11.2 by [com.puppycrawl.tools#checkstyle;6.15] in [default]
	---------------------------------------------------------------------
	|                  |            modules            ||   artifacts   |
	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
	---------------------------------------------------------------------
	|      default     |   13  |   0   |   0   |   1   ||   12  |   0   |
	---------------------------------------------------------------------
[ivy:retrieve] :: retrieving :: com.github.sevntu.checkstyle#sample
[ivy:retrieve] 	confs: [default]
[ivy:retrieve] 	0 artifacts copied, 12 already retrieved (0kB/24ms)

init-checkstyle:

checkstyle:
[checkstyle] Running Checkstyle 6.15 on 1 files

BUILD SUCCESSFUL
Total time: 1 second
```
With failOnViolation="true"
```
vlad@vlad-N61PC-M2S ~/projects/checkstyle-samples/ant-project $ ant checkstyle
Buildfile: /home/vlad/projects/checkstyle-samples/ant-project/build.xml

download-ivy:
      [get] Getting: http://repo2.maven.org/maven2/org/apache/ivy/ivy/2.1.0-rc2/ivy-2.1.0-rc2.jar
      [get] To: /home/vlad/projects/checkstyle-samples/ant-project/lib/ivy.jar
      [get] Not modified - so not downloaded

init-ivy:

resolve:
[ivy:retrieve] :: Ivy 2.1.0-rc2 - 20090704004254 :: http://ant.apache.org/ivy/ ::
[ivy:retrieve] :: loading settings :: file = /home/vlad/projects/checkstyle-samples/ant-project/ivysettings.xml
[ivy:retrieve] :: resolving dependencies :: com.github.sevntu.checkstyle#sample;working@vlad-N61PC-M2S
[ivy:retrieve] 	confs: [default]
[ivy:retrieve] 	found com.puppycrawl.tools#checkstyle;6.15 in local-maven-2
[ivy:retrieve] 	found org.apache.commons#commons-lang3;3.4 in local-maven-2
[ivy:retrieve] 	found antlr#antlr;2.7.7 in local-maven-2
[ivy:retrieve] 	found org.antlr#antlr4-runtime;4.5.1-1 in local-maven-2
[ivy:retrieve] 	found commons-beanutils#commons-beanutils;1.9.2 in local-maven-2
[ivy:retrieve] 	found commons-logging#commons-logging;1.1.1 in local-maven-2
[ivy:retrieve] 	found commons-collections#commons-collections;3.2.1 in local-maven-2
[ivy:retrieve] 	found commons-cli#commons-cli;1.3.1 in local-maven-2
[ivy:retrieve] 	found com.google.guava#guava;19.0 in local-maven-2
[ivy:retrieve] 	found org.apache.ant#ant;1.9.6 in local-maven-2
[ivy:retrieve] 	found org.apache.ant#ant-launcher;1.9.6 in local-maven-2
[ivy:retrieve] 	found com.github.sevntu.checkstyle#sevntu-checks;1.18.0 in local-maven-2
[ivy:retrieve] :: resolution report :: resolve 381ms :: artifacts dl 22ms
[ivy:retrieve] 	:: evicted modules:
[ivy:retrieve] 	com.puppycrawl.tools#checkstyle;6.11.2 by [com.puppycrawl.tools#checkstyle;6.15] in [default]
	---------------------------------------------------------------------
	|                  |            modules            ||   artifacts   |
	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
	---------------------------------------------------------------------
	|      default     |   13  |   0   |   0   |   1   ||   12  |   0   |
	---------------------------------------------------------------------
[ivy:retrieve] :: retrieving :: com.github.sevntu.checkstyle#sample
[ivy:retrieve] 	confs: [default]
[ivy:retrieve] 	0 artifacts copied, 12 already retrieved (0kB/16ms)

init-checkstyle:

checkstyle:
[checkstyle] Running Checkstyle 6.15 on 1 files

BUILD FAILED
/home/vlad/projects/checkstyle-samples/ant-project/build.xml:15: Got 1 errors and 0 warnings.

Total time: 1 second
```

**Comment by romani** [comment]
@Vladlis , to clear all further questions please add "git diff" output to prove how exactly build.xml was changed for `With failOnViolation="true/false"`. Please update your comment, do not create new comment.

**Comment by Vladlis** [comment]
@romani Done

**Comment by romani** [comment]
we can not reproduce the issue on latest code.

issue is closed
